sympy/polys/ring_series.py

```diff
@@ -1158,3 +1158,97 @@ def rs_compose_add(p1, p2):
     if dp:
         q = q*x**dp
     return q
+
+
+class RingSeriesBase(object):
+    pass
+
+class RingSeries(RingSeriesBase):
+    """Sparse Multivariate Ring Series"""
+
+    def new(self, *args, **kwargs):
+        return self.__class__(*args, **kwargs)
+
+    def __init__(cls, series, degree=None):
+        cls.series = series
+        cls.ring = series.ring
+        if degree:
+            cls.degree = degree
```

Comment:
``if degree`` can be removed. It will raise an error in ``get_degree`` if ``degree`` is ``None``.